src/Numerics/SoaCartesianTensor.h

```diff
@@ -140,6 +179,111 @@ SoaCartesianTensor<T>::SoaCartesianTensor(const int l_max, bool addsign) : Lmax(
 }
 
 
+PRAGMA_OFFLOAD("omp declare target")
+template<class T>
+void SoaCartesianTensor<T>::evaluate_bare_impl(T x, T y, T z, T* restrict XYZ, size_t Lmax_)
```

Comment:
Could you make `evaluate_bare` calling `evaluate_bare_impl` to minimize code duplication?

Reply:
fixed